test/bootstrap_rich_text_area_test.rb

```diff
@@ -8,18 +9,39 @@ class BootstrapRichTextAreaTest < ActionView::TestCase
     setup :setup_test_fixture
 
     test "rich text areas are wrapped correctly" do
-      expected = <<~HTML
-        <div class="mb-3">
-          <label class="form-label" for="user_life_story">Life story</label>
-          <input type="hidden" name="user[life_story]" id="user_life_story_trix_input_user"/>
+      with_stub_token do
```

Comment:
I must confess I'm too tired tonight to get my head around this test. I'll take another look tomorrow. Sorry.

Reply:
The `with_stub_token` is copied from a test in the Rails project itself (https://github.com/rails/rails/blob/main/actiontext/test/template/form_helper_test.rb) and makes sure that `data-direct-upload-token` is always the same value "token", so that we can validate the HTML output. Without this stub the value for `data-direct-upload-token` would be a big, random string.

On top of that, the markup for trim in Rails 7 differs in quite a few areas so I thought it would make sense to just make a big IF statement instead of doing everything inline and making a big(ger) mess.

Hope this makes sense. 😀

Reply:
Thanks for the explanation. It makes sense. See my comment a little further down.